### ISSUE TITLE
Preview for nvim bufferes is not showing.

### ISSUE THREAD

**Issue by Nithin745** [ibhagwan/fzf-lua#563]
### Info

- Operating System: Fedora 36
- Shell: 
- Terminal: 
- `nvim --version`: 0 .9
- `fzf --version`: 0.33.0

```lua
require('fzf-lua').setup({
    keymap = {
        fzf = {
            ["ctrl-q"] = "select-all+accept",
        },
    },
    -- previewers = {
    --     bat = {
    --         cmd = "bat",
    --         args = "--style=numbers,changes --color always",
    --         theme = 'Coldark-Dark',
    --     }
    -- },
    winopts = {
        height = 0.95,
        width = 0.95,
        preview = {
            scrollbar = false,
            default = 'bat_native'
        },
    },
    fzf_opts = {
        ["--layout"] = "default",
        -- ["--ansi"] = false, --makes fzf fast, but have some limitations
    },
    files = {
        -- cmd = 'find * -name ".git" -o -name "node_modules" -o -name "__pycache__" -prune -o -type f -print',
        -- find_opts         = [[-type f -not -path '*/\.git/*' -printf '%P\n']],
        rg_opts = "--color=never --files --hidden --follow -g '!.git'",
        fd_opts = "-t f -E node_modules -E venv -E __pycache__  -E venv",
        actions = {
            ["ctrl-e"] = function(selected)
                for i, item in ipairs(selected) do
                    local command = i == 1 and "edit" or i % 2 == 0 and "vsplit" or "split"
                    vim.cmd(string.format("%s %s", command, item))
                end
            end,
        },
    },

})
```
</details>

### Description 
while using `bat_native ` as a previewer, the preview for the opened buffer in nvim is not showing preview, instead its showning some error. if i use bat it showing the preview for the buffers as expected.
![Screenshot from 2022-11-06 14-53-02](https://user-images.githubusercontent.com/85417614/200163567-d0ea7999-70ed-42e9-906b-42f9cfb15bdc.png)
![Screenshot from 2022-11-06 14-53-02](https://user-images.githubusercontent.com/85417614/200163571-509c3803-6769-4a6f-bd76-478b69139058.png)

**Comment by ibhagwan** [comment]
What shell are you using? Does `bat` (not native) works?

**Comment by Nithin745** [comment]
> What shell are you using? Does `bat` (not native) works?

I'm using bash. Yes using `bat`  the config for fzf-lua instead of `bat_native` works just fine. I don't know why `bat_native` is not working.

**Comment by ibhagwan** [comment]
We can try and solve it but TBH there is no real benefit to bat native over bat, the native variant is only there for legacy reasons, the end result is also exactly the same which is a bat preview.

**Comment by Nithin745** [comment]
> We can try and solve it but TBH there is no real benefit to bat native over bat, the native variant is only there for legacy reasons, the end result is also exactly the same which is a bat preview.

Okay, i thought that `bat_native` was little bit faster that `bat`. But i think the problem is in the the command for the bat preview. Is there anyway i can print the the command, to the nvim terminal when i open the fuzzy finder for both normal files and buffers, so that i can compare.
if you want you can go ahead and close this issue, but just tell me if there is way to print the commands, i'm little bit curious

**Comment by ibhagwan** [comment]
You can run `:FzfLua buffers previewer=bat debug=true` and you'll be able to see the preview command in the preview window (before the bat output), unfortunately this isn't implemented for `bat_native` :)
![image](https://user-images.githubusercontent.com/59988195/200170786-69dc127d-8743-4ca9-958a-e4a637decd1e.png)

Let me see if I can fix it real quick.

**Comment by ibhagwan** [comment]
Just looked into it, `bat_native` never worked with buffers, I didn't even bother trying to fix it, the reason is that it runs the preview command, in our case `bat ... ` and the filename part is extracted from the entry via a `neovim --headless` call to the main instance, the full command being something like:
```sh
bat <args> `nvim  --headless ... <args> {}`
```
`{}` is fzf's field index expression which is exanded to fzf's currently selected line, this was causing many different issues in expanding certain characters into the backtick command and the main reason I replaced `bat_native` with `bat` (which doesn't use backticks in the preview command).

In the case of buffers you have all kinds of characters like `[]  % #` which cause these issues in the backtick expansion, TBH I don't know if it's worth the effort at this point.

**Comment by Nithin745** [comment]
> Just looked into it, `bat_native` never worked with buffers, I didn't even bother trying to fix it, the reason is that it runs the preview command, in our case `bat ... ` and the filename part is extracted from the entry via a `neovim --headless` call to the main instance, the full command being something like:
> 
> ```shell
> bat <args> `nvim  --headless ... <args> {}`
> ```
> 
> `{}` is fzf's field index expression which is exanded to fzf's currently selected line, this was causing many different issues in expanding certain characters into the backtick command and the main reason I replaced `bat_native` with `bat` (which doesn't use backticks in the preview command).
> 
> In the case of buffers you have all kinds of characters like `[] % #` which cause these issues in the backtick expansion, TBH I don't know if it's worth the effort at this point.

I agree with you, i just use `bat` preview for the buffers and `bat_native` for the rest of the fuzzy finder.
I really appreciate you looked into this so quickly, found the source of the issue as well, thanks.
And thanks for this incredible plugin :).

**Comment by ibhagwan** [comment]
>  agree with you, i just use `bat` preview for the buffers and `bat_native` for the rest of the fuzzy finder.
I really appreciate you looked into this so quickly, found the source of the issue as well, thanks.
And thanks for this incredible plugin :).

I just tried changing the backticks to `$(<cmd>)` instead, doesn't seem to make a big difference, fzf still has issues expanding the filename even when surrounding the `{}` with `"{}"`, still bugs me that something isn't working but probably not worth the effort.

Ty for your kind words :)

**Comment by ibhagwan** [comment]
FYI, fixed in https://github.com/ibhagwan/fzf-lua/commit/664358dc45eb4603284ddbfad308a454f6536638 so you can keep using `bat_native` for everything if you wish.

**Comment by Nithin745** [comment]
> FYI, fixed in [664358d](https://github.com/ibhagwan/fzf-lua/commit/664358dc45eb4603284ddbfad308a454f6536638) so you can keep using `bat_native` for everything if you wish.

Great, now works like charm now, thank you. I guess you really resist a chance to perfect everything :).
